pkg/deploy/deployment_che.go

```diff
@@ -54,6 +58,20 @@ func NewCheDeployment(cr *orgv1.CheCluster, cheImage string, cheTag string, cmRe
 			},
 		}
 	}
+	if cr.Spec.Server.CustomPublicCert {
+        customPublicCertEnv = corev1.EnvVar{
+            Name: "CHE_CUSTOM_PUBLIC_CERT",
+            ValueFrom: &corev1.EnvVarSource{
+                ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
+                    Key: "ca.crt",
+                    LocalObjectReference: corev1.LocalObjectReference{
+                        Name: "custom-public-cert",
+                    },
+                    Optional: &optionalEnv,
```

Comment:
I really don't get motivation why this env var should be optional. It may lead to an unpredictable situation when user configures `true` in CheCluster CR but does not create config map... Che Server will be redeployed but custom-public cert won't be propagated.
In my opinion, it's better to fail a reconciliation loop and tells that config map is missing

Reply:
fixed